scipy/_lib/_uarray/vectorcall.h

```diff
@@ -6,18 +6,16 @@ extern "C" {
 #endif
 
 // True if python supports vectorcall on custom classes
-#if (!defined(PYPY_VERSION) && (PY_VERSION_HEX >= 0x03080000))
+#if (!defined(PYPY_VERSION))
 #  define Q_HAS_VECTORCALL 1
 #else
 #  define Q_HAS_VECTORCALL 0
 #endif
```

Comment:
Now that there's no `&&`, we can lose the brackets and use `#ifndef` directly.
```suggestion
#ifndef PYPY_VERSION
#  define Q_HAS_VECTORCALL 1
#else
#  define Q_HAS_VECTORCALL 0
#endif
```

For consistency with the usage in `vectorcall.cxx`, we could also switch the cases around to avoid the inverted "not defined" logic (I have a slight preference for this):
```suggestion
#ifdef PYPY_VERSION
#  define Q_HAS_VECTORCALL 0
#else
#  define Q_HAS_VECTORCALL 1
#endif
```

Reply:
So in light of that, we should IMO do:
```suggestion
#ifdef PYPY_VERSION
#  define Q_Py_TPFLAGS_HAVE_VECTORCALL 0
#  define Q_Py_TPFLAGS_METHOD_DESCRIPTOR 0
#  define Q_PY_VECTORCALL_ARGUMENTS_OFFSET                                     \
    ((size_t)1 << (8 * sizeof(size_t) - 1))
#else
#  define Q_Py_TPFLAGS_HAVE_VECTORCALL Py_TPFLAGS_HAVE_VECTORCALL
#  define Q_Py_TPFLAGS_METHOD_DESCRIPTOR Py_TPFLAGS_METHOD_DESCRIPTOR
#  define Q_PY_VECTORCALL_ARGUMENTS_OFFSET PY_VECTORCALL_ARGUMENTS_OFFSET
#endif
```
and then delete all the `#if !Q_HAS_VECTORCALL` instances below.